test/e2e/upgrade_ironic_test.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func upgrade_ironic() {
+	clientSet := targetCluster.GetClientSet()
+	ironicDeployName := "baremetal-operator-ironic"
+	By("Upgrading ironic image based containers")
+	getIronicDeployment := func() (*appsv1.Deployment, error) {
+		return clientSet.AppsV1().Deployments("baremetal-operator-system").Get(ctx, ironicDeployName, metav1.GetOptions{})
+	}
+	deploy, err := getIronicDeployment()
+	Expect(err).To(BeNil())
+	for i := range deploy.Spec.Template.Spec.Containers {
```

Comment:
You can loop over the container structs themselves instead of using indexing.

Reply:
I think it is only possible to update the struct on the object using indexes? 
if you meant the following it wont work 

```
for _, container := range deploy.Spec.Template.Spec.Containers {
		switch container.Name {
		case
			"mariadb",
			"ironic-api",
			"ironic-dnsmasq",
			"ironic-conductor",
			"ironic-log-watch",
			"ironic-inspector":
			container.Image = "/metal3-io/ironic:latest"  // should be updated by index instead
		}
	}
```

Reply:
@macaptain check this